### PR TITLE
Add stow output json

### LINKED ISSUE #1632: Output objects' final location in json
Output json is required in the rule.
It should be correct, otherwise we get penalty.
- [x] pick task
- [x] stow task

### PR DESCRIPTION
**Pull request by knorth55**
Closes #1632

### CODE REVIEW

File: jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+            rospy.get_param('/left_hand/target_object')
+
+    def update_data(self):
+        is_updated = False
+        for arm in ['right', 'left']:
+            target_bin = self.arm_target_bin[arm]
+            target_object = self.arm_target_object[arm]
+            if target_object not in self.finished_object_list and \
+                    target_bin in 'abcdefghijkl' and \
+                    self.arm_state[arm] == ":place_object":
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: same comment.

**Comment by knorth55** [comment]
$ roscore
$ rosparam set /left_hand/target_bin :a
$ rosparam set /left_hand/state place_object
$ rosparam set /left_hand/target_object scotch_bubble_mailer
$ rosparam set /right_hand/state place_object
$ rosparam set /right_hand/target_bin :b
$ rosparam set /right_hand/target_object up_glucose_bottle

output json
https://gist.github.com/knorth55/bc58050f18713f0e1061766fa46e8c8a

**Comment by knorth55** [comment]
fixed

File: jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+    def update_param(self):
+        self.arm_state['right'] = rospy.get_param('/right_hand/state')
+        self.arm_target_bin['right'] = \
+            rospy.get_param('/right_hand/target_bin').replace(':', '')
+        self.arm_target_object['right'] = \
+            rospy.get_param('/right_hand/target_object')
+        self.arm_state['left'] = rospy.get_param('/left_hand/state')
+        self.arm_target_bin['left'] = \
+            rospy.get_param('/left_hand/target_bin').replace(':', '')
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: Maybe the replacement it not necessary.

File: jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+            if is_updated:
+                self.output_data['tote_contents'] = self.tote_contents
+                self.output_data['bin_contents'] = self.bin_contents
+                with open(self.output_path, 'w') as f:
+                    json.dump(self.output_data, f, sort_keys=True, indent=4)
+            rate.sleep()
+
+    def update_param(self):
+        self.arm_state['right'] = rospy.get_param('/right_hand/state')
+        self.arm_target_bin['right'] = \
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: same comment.

Comment (knorth55) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: ?

Comment (wkentaro) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: screen shot 2016-06-11 at 01 41 35

Comment (knorth55) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: i could not find any replacement here

Comment (wkentaro) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: You amended and the diff is gone, but I think there was below line:
```python
+            rospy.get_param('/right_hand/target_bin').replace(':', '')
```

Comment (knorth55) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: you pointed one line above the replacement.
i fixed it

File: jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+        self.arm_state = {}
+        self.arm_target_bin = {}
+        self.arm_target_object = {}
+        self.finished_object_list = []
+        with open(self.output_path, 'w') as f:
+            json.dump(self.output_data, f, sort_keys=True, indent=4)
+
+    def main(self):
+        rate = rospy.Rate(rospy.get_param('~rate', 1))
+        while not rospy.is_shutdown():
```
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: Please use timer.

File: jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py (hunk 3 above)
Comment (wkentaro) [inline jsk_2016_01_baxter_apc/node_scripts/output_json_stow.py]: Maybe that silly mistake could be corrected by your guess. :sob:

**Comment by knorth55** [comment]
fixed to use rospy.Timer

**Comment by knorth55** [comment]
add timeout case in `:verify_object` method

**Comment by wkentaro** [comment]
Why are there so many commits in this PR?
And there are some conflicts.
image

**Comment by knorth55** [comment]
Something wrong with rebasing.
U will fix it

**Comment by knorth55** [comment]
I will

**Comment by wkentaro** [comment]
This is pending because of missing object recognition in hand.

**Comment by knorth55** [comment]
related to #1795

**Pull request merged**